webapp/src/utils/validationUtils.ts

```diff
@@ -150,20 +153,44 @@ export function validatePassword(password: string): string | true {
   return true;
 }
 
+/**
+ * Validates a number against specified numerical limits.
+ *
+ * @param [options] - Configuration options for validation.
+ * @param [options.min=Number.MIN_SAFE_INTEGER] - Minimum allowed value.
+ * @param [options.max=Number.MAX_SAFE_INTEGER] - Maximum allowed value.
+ * @returns A function that accepts a number to validate, and returns true
+ * if validation is successful, or a localized error message if it fails.
+ */
+export function validateNumber(
+  options?: NumberValidationOptions,
+): (value: number) => string | true;
+
 /**
  * Validates a number against specified numerical limits.
  *
  * @param value - The number to validate.
- * @param options - Configuration options for validation including min and max values. (Optional)
- * @param [options.min=Number.MIN_SAFE_INTEGER] - Minimum allowed value for the number.
- * @param [options.max=Number.MAX_SAFE_INTEGER] - Maximum allowed value for the number.
+ * @param [options] - Configuration options for validation.
+ * @param [options.min=Number.MIN_SAFE_INTEGER] - Minimum allowed value.
+ * @param [options.max=Number.MAX_SAFE_INTEGER] - Maximum allowed value.
  * @returns True if validation is successful, or a localized error message if it fails.
  */
 export function validateNumber(
   value: number,
-  options?: ValidationOptions,
-): string | true {
-  if (typeof value !== "number" || isNaN(value) || !isFinite(value)) {
+  options?: NumberValidationOptions,
+): string | true;
+
+export function validateNumber(
+  valueOrOpts?: number | NumberValidationOptions,
+  options?: NumberValidationOptions,
+): (string | true) | ((value: number) => string | true) {
+  if (typeof valueOrOpts !== "number") {
+    return (v: number) => validateNumber(v, valueOrOpts);
+  }
+
+  const value = valueOrOpts;
+
+  if (isFinite(value)) {
```

Comment:
you're returning an error message when the number is valid (finite). This should be changed to check for the opposite condition.

Reply:
Fixed